### PR TITLE
bitswap: add `ledger` subcommand

### PR DESCRIPTION
**Pull request by thomas-gardner**
Bring on the Strategies.

### CODE REVIEW

**Comment by whyrusleeping** [comment]
please take a look at #2814

File: core/commands/bitswap.go [hunk 1]
```diff
@@ -154,12 +157,16 @@ var bitswapStatCmd = &cmds.Command{
 				return nil, u.ErrCast()
 			}
 			buf := new(bytes.Buffer)
-			fmt.Fprintln(buf, "bitswap status")
-			fmt.Fprintf(buf, "\tprovides buffer: %d / %d\n", out.ProvideBufLen, bitswap.HasBlockBufferSize)
-			fmt.Fprintf(buf, "\tblocks received: %d\n", out.BlocksReceived)
-			fmt.Fprintf(buf, "\tdup blocks received: %d\n", out.DupBlksReceived)
-			fmt.Fprintf(buf, "\tdup data received: %s\n", humanize.Bytes(out.DupDataReceived))
-			fmt.Fprintf(buf, "\twantlist [%d keys]\n", len(out.Wantlist))
+			fmt.Fprintf(buf, "bitswap status\n"+
```
Comment (RichardLitt) [inline core/commands/bitswap.go]: These should all be initially capitalized, I think.

**Comment by thomas-gardner** [comment]
Hi @whyrusleeping, is `findOrCreate` really necessary? If the ledger doesn't exist, is there any reason to have this command create one?

**Comment by whyrusleeping** [comment]
@thomas-gardner The find or create is necessary for the function of the bitswap code. Its an atomic operation to fetch the ledger for a peer (and create one while holding the lock if it doesnt exist).

Also, this PR is mostly a duplicate of #2814

**Comment by thomas-gardner** [comment]
@whyrusleeping---right, something about creating an unneeded ledger was
confusing to me, but I can see how using `findOrCreate` keeps things
tidy and atomic.

Re diff #2814, this PR doesn't export any of the `Bitswap` fields, & gets
the locking right now ( ...I hope.)

**Comment by whyrusleeping** [comment]
@thomas-gardner alright, if you could add a `ShortDescription` to the command and rebase this on master i'll get it merged in.

**Comment by jbenet** [comment]
Nice!

**Comment by jbenet** [comment]
We should move bitswap out into go-bitswap so it can get dev cycles faster
than in go-ipfs. Can write small tools to test it and leverage it so it can
evolve quickly and merge back into go-ipfs as needsd

**Comment by thomas-gardner** [comment]
Jeromy Johnson:
> @thomas-gardner alright, if you could add a `ShortDescription` to the
> command and rebase this on master i'll get it merged in.

Rebased & added.

**Comment by whyrusleeping** [comment]
alright, LGTM

**Pull request merged**